movement/watch_faces/complication/calculator_face.c

```diff
@@ -0,0 +1,263 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 <#author_name#>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+#include "calculator_face.h"
+
+void calculator_face_setup(movement_settings_t *settings, uint8_t watch_face_index, void ** context_ptr) {
+    (void) settings;
+    if (*context_ptr == NULL) {
+        *context_ptr = malloc(sizeof(calculator_state_t));
+        memset(*context_ptr, 0, sizeof(calculator_state_t));
+        // Do any one-time tasks in here; the inside of this conditional happens only at boot.
+    }
+    // Do any pin or peripheral setup here; this will be called whenever the watch wakes from deep sleep.
+}
+
+static void calculator_face_display_number(double num) {
```

Comment:
@joeycastillo how would you feel about a separate PR which put this function in the library? (i.e. watch_display_float)

Reply:
Not opposed to it, tho I will say that I think watch faces may want to have different strategies for displaying numbers; I'd like to think through it before offering a library function for it. For example, the location setting screen displays 40.73 degrees west as `- 40₇₃`, with the smaller numbers in the seconds place representing the decimal portion, whereas the temperature display shows 72.3°F as `72-3°F` (since there's no decimal segment, and the underscore doesn't work in all positions). 

Open to proposals on how to generalize this, but my instinct has been to have the library focus on displaying strings, and leave it up to individual watch faces how they want to format numbers into those strings.

Reply:
After playing around with the various methods this function uses to display numbers I think @joeycastillo might be right. The display is imho too constrained to really generalize this. Every watch face should proper document how they display their numbers and as long as it makes sense to the user, I think this is okay 👍🏽

Reply:
I totally agree it's not fully generalisable, but it still seemed like it might be a useful utility function that can handle displaying an arbitrary float. If you know more about the number (as in lat/long, temp) there's no reason to use it. Happy to keep it here, at any rate.